### PR TITLE
Cleanup domains after integration test

### PR DESCRIPTION
**Pull request by gdbelvin**
At the end of an integration test, soft-delete the domain and the corresponding Trillian trees. 

TODO: 
- Run doman and tree garbage collection.

### CODE REVIEW

**Comment by Martin2112** [comment]
Can we downgrade code coverage severity so it doesn't look like the build has failed?

File: core/adminserver/admin_server.go [hunk 1]
```diff
@@ -325,16 +325,42 @@ func (s *Server) initialize(ctx context.Context, logTree, mapTree *tpb.Tree) err
...
+		return nil, err
+	}
+
 	if err := s.domains.SetDelete(ctx, in.GetDomainId(), false); err != nil {
 		return nil, err
 	}
+
+	_, delLogErr := s.logAdmin.UndeleteTree(ctx, &tpb.UndeleteTreeRequest{TreeId: d.Log.TreeId})
+	_, delMapErr := s.mapAdmin.UndeleteTree(ctx, &tpb.UndeleteTreeRequest{TreeId: d.Map.TreeId})
+	if delLogErr != nil || delMapErr != nil {
```
Comment (Martin2112) [inline core/adminserver/admin_server.go]: One of these undeletions could fail, resulting in a weird state. Also SetDelete false is called before the trees are undeleted so they could both fail to undelete. Does that also create a bad state?

Comment (gdbelvin) [reply core/adminserver/admin_server.go]: If they failed, it would create a bad state. Should I try to undelete at that point?

**Comment by gdbelvin** [comment]
The code coverage warning is just making sure that code coverage doesn't diminish. So while I agree that it is annoying, I like having it to motivate more code coverage :-)

**Comment by Martin2112** [comment]
The code coverage thing is going to delay reviews because people will assume the build isn't working. I think it's a bad idea to make it fail the overall status.

**Pull request merged**